pylint/checkers/classes/class_checker.py

```diff
@@ -1776,85 +1777,94 @@ def _check_protected_attribute_access(
             Klass.
         """
         attrname = node.attrname
+        if not is_attr_protected(attrname):
+            return
+        if attrname in self.linter.config.exclude_protected:
+            return
+        inferred = safe_infer(node.expr)
+        if inferred and isinstance(inferred, (nodes.ClassDef, nodes.Module)):
+            if f"{inferred.name}.{attrname}" in self.linter.config.exclude_protected:
+                return
+        klass = node_frame_class(node)
+
+        # In classes, check we are not getting a parent method
+        # through the class object or through super
+        callee = node.expr.as_string()
+
+        # Typing annotations in function definitions can include protected members
+        if utils.is_node_in_type_annotation_context(node):
+            return
+
+        # We are not in a class, no remaining valid case
+        if klass is None:
+            self.add_message("protected-access", node=node, args=attrname)
+            return
+
+        # If the expression begins with a call to super, that's ok.
         if (
-            is_attr_protected(attrname)
-            and attrname not in self.linter.config.exclude_protected
+            isinstance(node.expr, nodes.Call)
+            and isinstance(node.expr.func, nodes.Name)
+            and node.expr.func.name == "super"
         ):
-            klass = node_frame_class(node)
+            return
 
-            # In classes, check we are not getting a parent method
-            # through the class object or through super
-            callee = node.expr.as_string()
+        # If the expression begins with a call to type(self), that's ok.
+        if self._is_type_self_call(node.expr):
+            return
 
-            # Typing annotations in function definitions can include protected members
-            if utils.is_node_in_type_annotation_context(node):
+        inferred = safe_infer(node.expr)
```

Comment:
Didn't we already infer that line 1784 ?

Reply:
Good one - I must have copied the code-added-by-me to this location but not removed it from where it was in the first commit. This commit is fully reverted now until the refactor MR is merged.